### PR TITLE
Get annotated gateways

### PR DESCRIPTION
**Pull request by rheaj99**
This PR adds an annotation to Istio Gateway to indicate that they should be handled by AGIC and creates a function to get all annotated gateways

### CODE REVIEW

File: pkg/annotations/ingress_annotations.go [hunk 1]
```diff
@@ -40,13 +41,17 @@ const (
 	// that this is an ingress resource meant for the application gateway ingress controller.
 	IngressClassKey = "kubernetes.io/ingress.class"
 
+	// IstioGatewayKey defines the key of the annotation which needs to be set in order to specify
+	// that this is sa gateway meant for the application gateway ingress controller.
```
Comment (akshaysngupta) [inline pkg/annotations/ingress_annotations.go]: typo

File: pkg/annotations/ingress_annotations.go [hunk 2]
```diff
@@ -40,13 +41,17 @@ const (
...
+
 	// ApplicationGatewayIngressClass defines the value of the `IngressClassKey` annotation that will tell the ingress controller
 	// whether it should act on this ingress resource or not.
 	ApplicationGatewayIngressClass = "azure/application-gateway"
 
 	// ApplicationGatewayIstioIngressClass defines the value of the `IngressClassKey` annotation that will tell the ingress controller whether
-	// this ingress resource is from Istio or not.
-	ApplicationGatewayIstioIngressClass = "istio-ingress"
+	// this gateway should be handled by AGIC or not
+	ApplicationGatewayIstioIngressClass = "controls-app-gw"
```
Comment (akshaysngupta) [inline pkg/annotations/ingress_annotations.go]: can we use the same ingress class for Istio ?

File: pkg/k8scontext/context.go [hunk 3]
```diff
@@ -347,6 +347,18 @@ func (c *Context) GetEndpointsForVirtualService(virtualService v1alpha3.VirtualS
 	}
 }
 
+// GetGateways returns all Istio Gateways that are annotated.
+func (c *Context) GetGateways(annotation string) []*v1alpha3.Gateway {
```
Comment (akshaysngupta) [inline pkg/k8scontext/context.go]: `annotation string` are we using this anywhere ?

File: pkg/k8scontext/context.go [hunk 4]
```diff
@@ -347,6 +347,18 @@ func (c *Context) GetEndpointsForVirtualService(virtualService v1alpha3.VirtualS
...
+	annotatedGateways := make([]*v1alpha3.Gateway, 0)
+	for _, gateway := range c.ListIstioGateways() {
+		annotated, _ := annotations.IsIstioGatewayIngress(gateway)
```
Comment (akshaysngupta) [inline pkg/k8scontext/context.go]:
```go
if annotated, _ := annotations.IsIstioGatewayIngress(gateway); annotated {
    annotatedGateways = append(annotatedGateways, gateway)
}
```

File: pkg/annotations/ingress_annotations.go (hunk 2 above)
Comment (rheaj99) [reply pkg/annotations/ingress_annotations.go]: I'm not sure - the Gateway I'm playing around with right now doesn't have an ingress class annotation, so I'm not sure if it's necessary, but I can add that in later if it is.

**Pull request closed without merging**